Fix linting, add DOCUMENTATION.md

NycoCC1993 opened the pull request:
Per the results of the 30 January 2024 CS398 "Course Schedulizer" interview among Professors VanderLinden and Pruim, as well as students Sun & Madza, therefore the following request asks to merge a version of the code which removes the more trivial linting errors and also adds a Documentation markdown file.

Intended path - "update/develop" -> "develop"

NycoCC1993 committed:
Removing lint errors

NycoCC1993 committed:
Add files via upload

kvlinden commented:
Please make the following mods:
- Remove the unneeded comments (e.g., "this line turns off eslint..."-the code is clear here already, "removed this import..."-can get this from the file edit history)
- Make sure to get rid of Harmony (e.g., appInterfaces.ts should not need to import Harmony). 
Thanks!

NycoCC1993 committed:
Removing extraneous alteration messages

SophiaSun18 committed:
Unused capacity-related sections removed

SophiaSun18 committed:
Term -> Academic Period

SophiaSun18 committed:
Department -> Parse the first piece of CourseSection

SophiaSun18 committed:
Prefix, Number, Section, Course Title -> Parse everything from Course Section, but didn't comment out previous cf functions

SophiaSun18 committed:
Faculty Hours, Student Hours -> Semester Hours

SophiaSun18 committed:
BuildingAndRoom -> Location 1 & 2, Using the same cf function but only creating meeting sections for 1

SophiaSun18 committed:
Meeting Day, Meeting Time -> Meeting Patterns

SophiaSun18 committed:
Split part of original courseSectionCallback() into courseCallback()
to fix the error of "cannot read properties of undefined"

SophiaSun18 committed:
SemesterStartDate, SemesterEndDate -> StartDate, EndDate
original StartDate (column name: TermStart) commented out: same value as SemesterStartDate, all combined into the StartDate column of new format
Semester Length = EndDate - StartDate

SophiaSun18 committed:
Fixed `readFullCSV.spec.ts` to pass `pnpm test` using updated example schedule

SophiaSun18 committed:
Fixed minor meetingTime issue to accomodate the most recent schedule

SophiaSun18 committed:
Update meetingPatternCallback() and timeCallback() to process sections with multiple meeting times

SophiaSun18 committed:
Update output format

SophiaSun18 committed:
Update testing script

SophiaSun18 committed:
Update deploy.yml for `production` deployment

NycoCC1993 committed:
Adding changes

NycoCC1993 committed:
Fix jest-dom version needs extend-expect

NycoCC1993 committed:
Merge pull request #349 from senior-knights/feature/update-scheduling-format

Feature/update scheduling format
